### ISSUE TITLE
``consider-using-f-string`` and ``logging-fstring-interpolation`` are mutually exclusive

### ISSUE THREAD

**Issue by angiarao** [pylint-dev/pylint#5286]
### Bug description

Enabling both C0209 (consider-using-f-string) and W1203(logging-fstring-interpolation) causes conflict issues.
If logging is using % formatting, then C0209 error message is throwing on the logging string. Is there a way to avoid this conflict?

### Configuration

```ini
# Maximum number of nested blocks for function / method body
max-nested-blocks=5

# Disable the message, report, category or checker with the given id(s). You
# can either give multiple identifiers separated by comma (,) or put this
# option multiple times (only on the command line, not in the configuration
# file where it should appear only once).You can also use "--disable=all" to
# disable everything first and then reenable specific checks. For example, if
# you want to run only the similarities checker, you can use "--disable=all
# --enable=similarities". If you want to run only the classes checker, but have
# no Warning level messages displayed, use"--disable=all --enable=classes
# --disable=W"
# Example:
# disable=E1608,W1627,E1601
# we are disabling all warnings as it would create a lot of noise
# we will enable only selected warnings when needed.
disable=all

# Enable the message, report, category or checker with the given id(s). You can
# either give multiple identifier separated by comma (,) or put this option
# multiple time. See also the "--disable" option for examples.
# List of all codes can be found here -
# https://pylint.pycqa.org/en/latest/technical_reference/features.html
# or http://pylint-messages.wikidot.com/all-codes


enable=C0114,C0206,C0209
    ,E0102,E1111,E1120,E1121,E1123,E1305,E1606,E1601
    ,E9004,E9005,E9006,E9007,E9008,E9009,E9010,E9011
    ,E9013,E9014,E9015,E9016
    ,R0912,R1702,R1734
    ,W0101,W0123,W0311,W0612,W0613,W0614,W1620
    ,W9001,W9002,W9003,W9017
```

### Command used

```shell
python3 -m pylint --rcfile=.py3lintrc --reports=no --msg-template="{line}|{column}|{msg_id}|{symbol}|{msg}" <filename>
```

### Pylint output

```shell
Advice  (C0209) pylint - Consider Using F String
Formatting a regular string which could be a f-string
>>>       62     log.debug('resource %s was discovered' % name)

Warning  (W1203) pylint - Logging Fstring Interpolation
Use lazy % or .format() or % formatting in logging functions
>>>       62     log.debug(f'resource {name} was discovered')
```

### Expected behavior

i think C0209 should be suppressed if the W1203 is enabled. 

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.4
Python 3.6.9 (default, Jan 26 2021, 15:33:00)
[GCC 8.4.0]
```

### OS / Environment

Ubuntu 18.04

**Comment by Pierre-Sassoulas** [comment]
Could probably be a configuration warning so users need to take a decision themselves.

**Comment by angiarao** [comment]
the problem is for newbie developers and large diffs, it creates a lot of noise. It would be great to disable fstring check for logging statements if W1203 is enabled.

**Comment by Pierre-Sassoulas** [comment]
I like this, if one of them is explicitly enabled then we disable the other one, if both of them are enabled then we raise a warning.

**Comment by fmigneault** [comment]
C0209 seems to be a bad side effect here. It should simply be W1203.

Both of these definitions are incorrect according to best practices for logging package:

```
Advice  (C0209) pylint - Consider Using F String
Formatting a regular string which could be a f-string
>>>       62     log.debug('resource %s was discovered' % name)

Warning  (W1203) pylint - Logging Fstring Interpolation
Use lazy % or .format() or % formatting in logging functions
>>>       62     log.debug(f'resource {name} was discovered')
```

The correct fix is to use:
```python
log.debug('resource %s was discovered', name)
```
This is the optimal way because, contrary to the other 2 approaches, passing the parameter allows logging to skip any string formatting if the logger is not `level <= DEBUG`. The 2 erroneous approaches waste time performing string formatting before logging, even if the message does not even get logged when `level > DEBUG`.

**Comment by Pierre-Sassoulas** [comment]
Not wanting to enable ``W1203`` is a user choice, f-strings are efficient and looks better, and the performance increase is low. There was much vitriol in the discussion about this in another issue, so we're going to keep allowing both possibilities. Beside if we want to follow KISS, pylint should not emit message conditionally depending on other message being activated or not.

For someone wanting to implement this it would be nice to have a data structure with mutually exclusive msgid and raise ``mutually-exclusive-option-value`` when we encounter two of them activated at once.

**Comment by fmigneault** [comment]
A situation where performance could matter a lot is when the passed argument is a class that implements `__str__` or `__repr__` to delegate a resource intensive or long computation to be displayed only as needed. Usual string formatting that are proposed would always perform the computation, while the logging argument approach would not if the level is too low. 

In that situation, I would argue that pylint is not proposing the *optimal approach*. If a user was not aware of this optimization, and decided to update their code with `"%s"` or `f"{}"` formatting, they could unknowingly degrade the performance of their code. This is problematic because pylint should be (more or less) only affecting the code aesthetically without noticeable performance hit.

Since pylint is proposing changes, it should strive for the best practices. Detecting that logging is involved in those specific cases should be relatively easy because logging module methods have specific names.

**Comment by Pierre-Sassoulas** [comment]
We're allowing both options you just have to change the configuration if you want lazy logging. fstring are better in most project and most situation (and more popular than lazy logging by far). So It's going to stay the default.

**Comment by clavedeluna** [comment]
@Pierre-Sassoulas what's the action item here at this time?

**Comment by Pierre-Sassoulas** [comment]
Create a generic handling for mutually exclusive message that will raise 'mutually-exclusive-option-value' for the configuration file. There's already bad-option-value and useless-option-value that are pretty close to get some inspiration and guidance about how to do that (where the check should be done and how to create a configuration functional tests in particular). Then add those two messages to it.

**Comment by prerna0412** [comment]
Hi! I'd like to work on this issue for Hacktoberfest. Please assign me if it's available
